src/main/java/tasks/Event.java

```diff
@@ -0,0 +1,126 @@
+package tasks;
+
+import static java.lang.Integer.parseInt;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import exceptions.InvalidDateException;
+import exceptions.InvalidTaskNameException;
+
+
+
+
+/**
+ * Represents the Event task.
+ * A <code>Event</code> task has a name, from date, to date and a flag to show if it is done
+ */
+public class Event extends Task {
+
+    protected LocalDate fromDate;
+    protected LocalDate toDate;
+
+    /**
+     * Constructor for an Event task by taking in a string with the relevant information.
+     * It includes the name, from date and to date of the Event task.
+     *
+     * @param inputString The string containing information of the task
+     * @throws InvalidTaskNameException If no name is provided.
+     * @throws InvalidDateException If invalid date is provided.
+     *
+     */
+    public Event(String inputString) throws InvalidTaskNameException, InvalidDateException {
+
+        if (inputString.contains("/from ")) {
+            int fromIndex = inputString.indexOf("/from ");
+            String taskName = inputString.substring(0, fromIndex).trim();
+            if (taskName.length() == 0) {
+                throw new InvalidTaskNameException();
+            }
+
+            if (inputString.contains("/to ")) {
+                int toIndex = inputString.indexOf("/to ");
+                if (toIndex < fromIndex) {
+                    throw new InvalidDateException("Wrong date order provided!");
+                }
+                String fromDate = inputString.substring(fromIndex + 6, toIndex);
+                String toDate = inputString.substring(toIndex + 4);
+                if (fromDate.length() == 0) {
+                    throw new InvalidDateException("From date not provided");
+                } else if (toDate.length() == 0) {
+                    throw new InvalidDateException("To date not provided");
+                }
+                this.name = taskName;
+                try {
+                    this.fromDate = LocalDate.parse(fromDate.trim());
+                } catch (DateTimeParseException ex) {
+                    throw new InvalidDateException("Invalid from date format given");
+                }
+
+                try {
+                    this.toDate = LocalDate.parse(toDate.trim());
+                } catch (DateTimeParseException ex) {
+                    throw new InvalidDateException("Invalid to date format given");
+                }
+
+                if (this.toDate.isBefore(this.fromDate)) {
+                    throw new InvalidDateException("To date is before from date");
+                }
+            } else {
+                throw new InvalidDateException("To date is not provided!");
+            }
+        } else {
+            throw new InvalidDateException("From date is not provided!");
+        }
+    }
```

Comment:
I'm not sure I like how the line breaks were placed for this method. Perhaps placing a few more line breaks (e.g. in lines 46, 48, 53) could help with readability of the code by segmenting it into smaller sections.